### PR TITLE
add the primary indices to the AQL modification nodes serialization and their explain output

### PR DESCRIPTION
**Pull request by dothebart**
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-pr/20/

### CODE REVIEW

File: js/common/modules/@arangodb/aql/explainer.js [hunk 1]
```diff
@@ -992,10 +992,30 @@ function processQuery (query, explain) {
...
+    var what = (node.reverse ? 'reverse ' : '') + idx.type + ' index scan' + ((node.producesResult || !node.hasOwnProperty('producesResult')) ? (node.indexCoversProjections ? ', index only' : '') : ', scan only');
+    if (types.length === 0 || what !== types[types.length - 1]) {
+      types.push(what);
+    }
+    idx.collection = node.collection;
+    idx.node = node.id;
+    if (node.hasOwnProperty('condition') && node.condition.type && node.condition.type === 'n-ary or') {
+      idx.condition = buildExpression(node.condition.subNodes[i]);
+    } else {
+      if (variable !== false) {
```
Comment (jsteemann) [inline js/common/modules/@arangodb/aql/explainer.js]: Why `false` here?
If the function is called with just 4 parameters, `variable` will be `undefined`, not `false`

File: arangod/Aql/CollectionAccessingNode.cpp [hunk 2]
```diff
@@ -83,3 +84,12 @@ void CollectionAccessingNode::toVelocyPack(arangodb::velocypack::Builder& builde
     builder.add("restrictedTo", VPackValue(_restrictedTo));
   }
 }
+
+void CollectionAccessingNode::toVelocyPackHelperPrimaryIndex(arangodb::velocypack::Builder& builder) const {
+  auto col = _collection->getCollection();
+  builder.add(VPackValue("indexes"));
+  col->getIndexesVPack(builder, false, false, [](arangodb::Index const* idx) {
+      return (idx->type() == arangodb::Index::TRI_IDX_TYPE_PRIMARY_INDEX ||
+              idx->type() == arangodb::Index::TRI_IDX_TYPE_EDGE_INDEX);
```
Comment (jsteemann) [inline arangod/Aql/CollectionAccessingNode.cpp]: it's very confusing that this method is supposed to return the VPack for the primary index, but will apparently also do so for the edge index. This makes no sense.

File: js/common/modules/@arangodb/aql/explainer.js [hunk 3]
```diff
@@ -992,10 +992,30 @@ function processQuery (query, explain) {
     }
     return '';
   };
+  
+  var iterateIndices = function (idx, i, node, types, variable) {
```
Comment (jsteemann) [inline js/common/modules/@arangodb/aql/explainer.js]: can we rename this to `iterateIndexes` please?

File: js/common/modules/@arangodb/aql/explainer.js (hunk 1 above)
Comment (dothebart) [reply js/common/modules/@arangodb/aql/explainer.js]: yes, false is missing here.

**Pull request merged**